Userland/Libraries/LibGUI/ComboBox.cpp

```diff
@@ -293,6 +293,10 @@ ByteString ComboBox::text() const
 void ComboBox::set_text(ByteString const& text, AllowCallback allow_callback)
 {
     m_editor->set_text(text, allow_callback);
+    if (!on_change)
+        return;
+    if (auto match = model()->matches(text.view(), GUI::Model::MatchesFlag::FirstMatchOnly); match.size() == 1)
+        on_change(text, match[0]);
```

Comment:
Should we pay attention to `allow_callback` for `on_change` too? I'm not sure of the implications of that, but it feels weird to ignore it.

Reply:
allow_callback was a no-op before, so I stopped plumbing it through to the internal TextEditor and started checking it here to control if ComboBox::on_change is called. This is what we discussed on discord